Wikipedia/UI-V5/UIViewController+WMFSearchButton.m

```diff
@@ -0,0 +1,51 @@
+//
+//  UIViewController+WMFSearchButton.m
+//  Wikipedia
+//
+//  Created by Brian Gerstle on 9/30/15.
+//  Copyright © 2015 Wikimedia Foundation. All rights reserved.
+//
+
+#import "UIViewController+WMFSearchButton.h"
+#import "WMFSearchViewController.h"
+#import "UIBarButtonItem+WMFButtonConvenience.h"
+#import "WMFArticleContainerViewController.h"
+#import "SessionSingleton.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+@implementation UIViewController (WMFSearchButton)
+
+- (UIBarButtonItem*)wmf_searchBarButtonItemWithDelegate:(UIViewController<WMFSearchPresentationDelegate>*)delegate {
+    @weakify(self);
+    @weakify(delegate);
+    return [UIBarButtonItem wmf_buttonType:WMFButtonTypeMagnify handler:^(id sender) {
+        @strongify(self);
+        @strongify(delegate);
+        if (!delegate || !self) {
+            return;
+        }
+
+        MWKSite* searchSite;
+        if ([delegate respondsToSelector:@selector(searchSite)]) {
```

Comment:
Sorry, my question is why do we have a search delegate at all - why are we not deferring to the app wide setting?